src/modules/routing/services/ProviderRoutingService.ts

```diff
@@ -12,11 +18,15 @@ export interface RoutingTable {
 class ProviderRoutingService {
   private routingTable: RoutingTable = {}
 
-  /**
-   * @todo use connection from message context
-   */
-  public updateRoutes(messageContext: InboundMessageContext<KeylistUpdateMessage>, connection: ConnectionRecord) {
-    const { message } = messageContext
+  public updateRoutes(messageContext: InboundMessageContext<KeylistUpdateMessage>): KeylistUpdateResponseMessage {
+    const { connection, message } = messageContext
+
+    if (!connection) {
+      // TODO We could eventually remove this check if we do it at some higher level where we create messageContext that must have a connection.
```

Comment:
Yup would be nice to be able to declare if a connection is required for a handler or something...